src/OpenTelemetry/Metrics/PullMetricScope.cs

```diff
@@ -0,0 +1,53 @@
+// <copyright file="PullMetricScope.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Runtime.CompilerServices;
+using OpenTelemetry.Context;
+
+namespace OpenTelemetry
+{
+    internal sealed class PullMetricScope : IDisposable
```

Comment:
I like this as a way for exporter to tell `Collect` that it wants to allow pull. Consistent with how we do the instrumentation suppression.